src/components/app/content/index.js

```diff
@@ -1,12 +1,38 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './index.css';
-import Card from "./card";
+import Cards from "./cards";
 
 function Content() {
+    const [readOnly, setReadOnly] = useState(false)
+    const [cards, setCards] = useState([
+        {id: 1, header: 'Caption', body: 'Text...', checked: false, editable: false},
+        {id: 2, header: 'Caption', body: 'Text...', checked: false, editable: false},
+        {id: 3, header: 'Caption', body: 'Text...', checked: false, editable: false},
+        {id: 4, header: 'Caption', body: 'Text...', checked: false, editable: false},
+        {id: 5, header: 'Caption', body: 'Text...', checked: false, editable: false},
+        {id: 6, header: 'Caption', body: 'Text...', checked: false, editable: false},
+        {id: 7, header: 'Caption', body: 'Text...', checked: false, editable: false},
+        {id: 8, header: 'Caption', body: 'Text...', checked: false, editable: false},
+    ])
+
+    const editCard = (cardId, cardData) => {
+        setCards(
+            cards.map(card => {
+                if (card.id === cardId) {
+                    card = cardData
```

Comment:
What if card entry contains more properties than are passed? Better use spread operator to copy existing and merge with provided.
Also. When using map - you're iterating throw the whole array with applying/ not applying changes. No sense.
Better get the required card entry by find method and the and then merge changes.